src/main/java/org/fenixedu/academic/thesis/domain/ThesisProposalParticipant.java

```diff
@@ -54,6 +67,8 @@ public void delete() {
         this.setUser(null);
         this.setThesisProposal(null);
 
+        this.setExternalUser(null);
```

Comment:
If ExternalUser instances are not shared they should be deleted when the participation is deleted